File: pippip-native-ios/Support/NotificationErrorDelegate.m
//
//  NotificationErrorDelegate.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 3/21/18.
//  Copyright © 2018 seComm. All rights reserved.
//

#import "NotificationErrorDelegate.h"

@interface NotificationErrorDelegate ()
{
    UIAlertController *alert;
    NSMutableDictionary *info;
}

@end

@implementation NotificationErrorDelegate

- (instancetype)initWithTitle:(NSString*)title {
    self = [super init];

    alert = [UIAlertController alertControllerWithTitle:title
                                                message:@""
                                         preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
                                                       style:UIAlertActionStyleDefault
                                                     handler:nil];
    [alert addAction:okAction];
    info = [NSMutableDictionary dictionary];
    info[@"alert"] = alert;

    return self;

}

- (void)getMethodError:(NSString*)error {

    alert.message = error;
    [[NSNotificationCenter defaultCenter] postNotificationName:@"PresentAlert" object:nil userInfo:info];

}

- (void)postMethodError:(NSString*)error {

    alert.message = error;
    [[NSNotificationCenter defaultCenter] postNotificationName:@"PresentAlert" object:nil userInfo:info];

}

- (void)responseError:(NSString*)error {

    alert.message = error;
    [[NSNotificationCenter defaultCenter] postNotificationName:@"PresentAlert" object:nil userInfo:info];

}

- (void)sessionError:(NSString*)error {

    alert.message = error;
    [[NSNotificationCenter defaultCenter] postNotificationName:@"PresentAlert" object:nil userInfo:info];

}

@end

File: pippip-native-ios/Support/NotificationErrorDelegate.swift
//
//  NotificationErrorDelegate.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 3/23/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit

@objc class NotificationErrorDelegate: NSObject, ErrorDelegate {

    var alertPresenter = AlertPresenter()
    var title: String

    @objc init(title: String) {

        self.title = title

    }

    func getMethodError(_ error: String) {

        alertPresenter.errorAlert(title: title, message: error)
        
    }
    
    func postMethodError(_ error: String) {

        alertPresenter.errorAlert(title: title, message: error)
        
    }

    func responseError(_ error: String) {

        alertPresenter.errorAlert(title: title, message: error)
        
    }
    
    func requestError(_ error: String) {

        alertPresenter.errorAlert(title: title, message: error)
        
    }
    
    func sessionError(_ error: String) {

        alertPresenter.errorAlert(title: title, message: error)
        
    }
    
}
